test/new-e2e/utils/e2e/e2e.go

```diff
@@ -0,0 +1,225 @@
+	suite.add(config.DDInfraConfigNamespace, aws.DDInfraDefaultKeyPairParamName, "agent-ci-sandbox")
+	suite.add(config.DDInfraConfigNamespace, aws.DDInfraDefaultInstanceTypeParamName, "t3.large")
+	suite.add(config.DDInfraConfigNamespace, aws.DDInfraDefaultARMInstanceTypeParamName, "m6g.medium")
+}
+
+func (c *Suite) add(namespace string, key string, value string) {
+	c.configMap[namespace+":"+key] = auto.ConfigValue{Value: value}
+}
+
+// TearDownSuite method is run after all the tests in the suite have been run.
+// This function is called by [testify Suite].
+//
+// [testify Suite]: https://pkg.go.dev/github.com/stretchr/testify/suite
+func (suite *Suite) TearDownSuite() {
+	if suite.fullStackName != "" {
+		var err error
+		if suite.destroyEnv {
+			err = infra.GetStackManager().DeleteStack(suite.ctx, suite.envName, suite.stackName)
+		}
+
+		if !suite.destroyEnv || err != nil {
+			stars := strings.Repeat("*", 50)
+			thisFolder := "A_FOLDER_CONTAINING_PULUMI.YAML"
+			if _, thisFile, _, ok := runtime.Caller(0); ok {
+				thisFolder = filepath.Dir(thisFile)
+			}
+
+			command := fmt.Sprintf("pulumi destroy -C %v --remove  -s %v", thisFolder, suite.fullStackName)
+			fmt.Fprintf(os.Stderr, "\n%v\nYour environment was not destroyed.\nTo destroy it, run `%v`.\n%v", stars, command, stars)
+		}
+		suite.require.NoError(err)
+	}
+}
+
+// Client provides a client to interact with the environment.
+type Client struct {
+	require   *require.Assertions
+	sshClient *ssh.Client
+}
+
+// Execute executes a command on the remote environment.
+func (c *Client) Execute(command string) string {
+	output, err := clients.ExecuteCommand(c.sshClient, command)
+	c.require.NoError(err, "Command `%v` failed with the output `%v`", command, output)
+	return output
+}
+
+// Run creates an environment and runs the E2E test.
+// The first argument is a function that creates the environment. It must return the type ([*vm.VM], error).
+// The second argument is a function that implements the E2E test logic.
+func (suite *Suite) Run(envFactory func(ctx *pulumi.Context) (*vm.VM, error), testFct func(*Client)) {
+	sshClient, err := suite.createEnv(envFactory)
+	suite.require.NoError(err)
+	defer sshClient.Close()
+	client := &Client{require: suite.require, sshClient: sshClient}
+	testFct(client)
+}
+
+func (suite *Suite) createEnv(fct func(ctx *pulumi.Context) (*vm.VM, error)) (*ssh.Client, error) {
+	stack, stackOutput, err := infra.GetStackManager().GetStack(suite.ctx, suite.envName, suite.stackName, suite.configMap, func(ctx *pulumi.Context) error {
+		_, err := fct(ctx)
+		return err
+	})
+	if err != nil {
+		return nil, err
+	}
+	if suite.fullStackName == "" {
+		suite.fullStackName = stack.Name()
+	}
+
+	suite.require.Equal(suite.fullStackName, stack.Name())
+	user, ip, err := getConnection(stackOutput)
+	if err != nil {
+		return nil, fmt.Errorf("your environment doesn't have a valid virtual machine: %v", err)
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement. -->
**🥜 nitpick**
```suggestion
		return nil, fmt.Errorf("error connecting to your environment: %v", err)
```